### PR TITLE
Docs: Restructure Search content

### PR DESCRIPTION
**Pull request by imatwawana**
This PR:

- Creates a new **Search dashboards** page under **Dashboards**
- Moves dashboards and folder-related search content to that page
- Updates content on existing **Search** page

### CODE REVIEW

File: docs/sources/search/_index.md [hunk 1]
```diff
@@ -14,111 +16,24 @@ title: Search
...
 ## Command palette
 
-The command palette enables you to:
+The command palette allows you to:
 
 - Search for and open dashboards and folders
 - Create dashboards and alert rules
 - Locate pages within Grafana
 - Change the theme to dark or light
+- Run Sift investigations
```
Comment (eledobleefe) [inline docs/sources/search/_index.md:31]: I did not know about this one. I checked in ops, and I found it using the command palette, but [in the documentation](https://grafana.com/docs/grafana-cloud/alerting-and-irm/machine-learning/sift/sift/#sift-investigations), we don't have the command palette as a way to run them. Is that on purpose?

**Review by eledobleefe — APPROVED** [review]
Just had one comment, otherwise LGTM 🎉

**Review by irenerl24 — APPROVED** [review]
LGTM! Thank you!

File: docs/sources/search/_index.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/search/_index.md:31]: > I did not know about this one. I checked in ops, and I found it using the command palette, but [in the documentation](https://grafana.com/docs/grafana-cloud/alerting-and-irm/machine-learning/sift/sift/#sift-investigations), we don't have the command palette as a way to run them. Is that on purpose?

Good question! @alyssawada, do you have any insight here?

Comment (imatwawana) [reply docs/sources/search/_index.md:31]: > > I did not know about this one. I checked in ops, and I found it using the command palette, but [in the documentation](https://grafana.com/docs/grafana-cloud/alerting-and-irm/machine-learning/sift/sift/#sift-investigations), we don't have the command palette as a way to run them. Is that on purpose?
> 
> Good question! @alyssawada, do you have any insight here?

I confirmed with @JohnnyK-Grafana (who works on the machine learning content) that you can indeed launch a sift investigation from the command palette. He's going to update those docs. So we're all good!

**Pull request merged**